packages/mask/src/plugins/MaskBox/hooks/useContext.ts

```diff
@@ -25,31 +25,40 @@ import {
     ZERO_ADDRESS,
     isZeroAddress,
     isNativeTokenAddress,
+    formatBalance,
 } from '@masknet/web3-shared-evm'
 import type { NonPayableTx } from '@masknet/web3-contracts/types/types'
 import { BoxInfo, BoxState } from '../type'
 import { useMaskBoxInfo } from './useMaskBoxInfo'
-import { useGetRootHash } from './useGetRootHash'
+import { useMerkelProof } from './useMerkleProof'
 import { useMaskBoxStatus } from './useMaskBoxStatus'
 import { useMaskBoxCreationSuccessEvent } from './useMaskBoxCreationSuccessEvent'
 import { useMaskBoxTokensForSale } from './useMaskBoxTokensForSale'
 import { useMaskBoxPurchasedTokens } from './useMaskBoxPurchasedTokens'
 import { formatCountdown } from '../helpers/formatCountdown'
 import { useOpenBoxTransaction } from './useOpenBoxTransaction'
 import { useMaskBoxMetadata } from './useMaskBoxMetadata'
-import { useIsWhitelisted } from './useIsWhitelisted'
-import { isGreaterThanOrEqualTo, isLessThanOrEqualTo, isZero, multipliedBy, useBeat } from '@masknet/web3-shared-base'
-
-function useContext(initialState?: { boxId: string; qualification: string }) {
+import { useQualification } from './useQualification'
+import {
+    isGreaterThan,
+    isGreaterThanOrEqualTo,
+    isLessThanOrEqualTo,
+    isZero,
+    multipliedBy,
+    useBeat,
+} from '@masknet/web3-shared-base'
+
+function useContext(initialState?: { boxId: string; hashRoot: string }) {
     const now = new Date()
     const beat = useBeat()
     const account = useAccount()
     const chainId = useChainId()
     const { NATIVE_TOKEN_ADDRESS } = useTokenConstants(ChainId.Mainnet)
     const { MASK_BOX_CONTRACT_ADDRESS } = useMaskBoxConstants()
+    const coder = ABICoder as unknown as ABICoder.AbiCoder
 
     const [boxId, setBoxId] = useState(initialState?.boxId ?? '')
-    const qualification = initialState?.qualification?.replace(/0x/, '')
+    const [hashRoot, setHashRoot] = useState(initialState?.hashRoot ?? '')
```

Comment:
```suggestion
    const [hashRoot, setHashRoot] = useState(initialState?.hashRoot || '')
```
`||`  is safer than `??` at runtime.